core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java

```diff
@@ -186,6 +190,46 @@ public void testCloudSqlInstanceForcesRefresh() throws Exception {
     assertThat(refreshCount.get()).isEqualTo(2);
   }
 
+  @Test
+  public void testCloudSqlInstanceRetriesOnInitialFailures() throws Exception {
+    InstanceData data =
+        new InstanceData(
+            null, new SslData(null, null, null), Instant.now().plus(1, ChronoUnit.HOURS));
```

Comment:
Should we introduce test helpers to create these objects? All these nulls make it hard to see what we're doing otherwise.

Reply:
Good idea. Done.